lib/util/ImpCentralApiHelper.js

```diff
@@ -428,15 +454,24 @@ class ImpCentralApiHelper {
         return result;
     }
 
-    _processImpCentralApiRequest(entityType, id, impCentralApiMethod, ...args) {
+    _processImpCentralApiRequest(refreshAccessTokenOnAuthError, entityType, id, impCentralApiMethod, ...args) {
         UserInteractor.spinnerStart();
         return impCentralApiMethod(...args).
             then(result => this._resolveImpCentralApiResponse(result)).
             catch(error => {
-                if (this._isRateLimitError(error)) {
+                if (this._isAuthenticationError(error)) {
+                    if (refreshAccessTokenOnAuthError) {
+                        // try to refresh access token in case of authentication error and then restart the original request
+                        return this._refreshAccessToken().
+                            then(() => this._processImpCentralApiRequest(
+                                refreshAccessTokenOnAuthError, entityType, id, impCentralApiMethod, ...args));
```

Comment:
shouldn't it be `false` instead of `refreshAccessTokenOnAuthError` here (as well as below)?

Reply:
I think no. false here means that we will try to refresh expired access token only once per request.
But theoretically access tokens can expire at any moment, eg just after the previous refresh.
Why not to eliminate this case using multiple refresh per request?

Reply:
I see ok. You are right.
The only concern that I initially had is that it (in theory) may take forever. Which is probably acceptable at this point.